### PR TITLE
hotfix/WP-232-234-APCD-frontend-form-sanitation-PROD

### PR DESCRIPTION
**Pull request by sophia-massie**
## Overview
The purpose of this hotfix is to update the prod branch so that forms still allow special characters that users may use but restricting invalid characters on form submissions. 
Your thoroughness of testing is really appreciated! I know it's tedious. Using Django or React utilities in place of this band-aid is the long term goal once the fire is put out.

## Related

- [WP-232](https://jira.tacc.utexas.edu/browse/WP-232)
- [WP-233](https://jira.tacc.utexas.edu/browse/WP-233)
- [WP-234](https://jira.tacc.utexas.edu/browse/WP-234)
- mimics #195

## Changes

The checkForBlankInputs javascript was changed using whitelist methods to allow characters in input. At the same time, the script blacklists some explicit scripting character patterns. This is a front end change to prevent bad characters from making it to the server. The DB utils remove bad characters server side in apcd_database.py. 

## Testing

1. Go to [http://localhost:8000/register/request-to-submit/](http://localhost:8000/register/request-to-submit/)
2. Try to input **<script> </script> javascript.exe, anything in "quotes" or 'quotes" { hello.exe } thisshouldwork@test.com into any field** into a field
3. On change. the field should be replaced  with **Try to input script script exe anything in quotes or quotes  helloexe  thisshouldworktestcom**
4. Go to [http://localhost:8000/submissions/threshold-exception/](http://localhost:8000/submissions/threshold-exception/)
5. Repeat steps 2 and 3
6. Go to [http://localhost:8000/submissions/other-exception/](http://localhost:8000/submissions/other-exception/)
7. Repeat steps 2 and 3
8. Go to [https://apcd-qa.tacc.utexas.edu/submissions/extension-request/](https://apcd-qa.tacc.utexas.edu/submissions/extension-request/)
9.  Repeat steps 2 and 3

### User input:

Screenshot 2023-08-22 at 6 07 51 PM

### On change, text is changed to:

Screenshot 2023-08-22 at 6 08 53 PM

### CODE REVIEW

File: apcd-cms/src/apps/registrations/templates/submission_form/registration_form_scripts.html [hunk 1]
```diff
@@ -1,4 +1,12 @@
-<aside>
+<aside>    
+  <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
```
Comment (wesleyboar) [inline apcd-cms/src/apps/registrations/templates/submission_form/registration_form_scripts.html:4]: This jQuery load seems like it doesn't need to be copied over. This page does not have the jQuery that https://github.com/TACC/Core-CMS-Custom/pull/195/files#diff-8555cd54ebd055a9434e6c60c435fe58b59ecc6facdad1fde497a0b7b97d7c72R152-R153 does.

<sup>P.S. [jQuery (at an older version) is already loaded by Core-CMS](https://github.com/TACC/Core-CMS/blob/v3.11.6/taccsite_cms/templates/assets_core_delayed.html#L6).</sup>
<sup>P.P.S. [The jQuery code](https://github.com/TACC/Core-CMS/blob/v3.11.6/taccsite_cms/templates/assets_core_delayed.html#L6) could be written without jQuery with minimal changes. Out of scope, I know.</sup>

Comment (sophia-massie) [reply apcd-cms/src/apps/registrations/templates/submission_form/registration_form_scripts.html]: Updated and removed!

**Review by chandra-tacc — APPROVED** [review]

File: apcd-cms/src/apps/utils/js/checkforBlankInputs.js
Comment (wesleyboar) [inline apcd-cms/src/apps/utils/js/checkforBlankInputs.js:1]: This script does not load. The capitalization in the name is not the only problem.

**Review by wesleyboar — CHANGES_REQUESTED** [review]
I can test [/register/request-to-submit/](http://localhost:8000/register/request-to-submit/). Script does not load. I have a fix. Expect PR.

Every other page crashes or is empty. I assume my local env. is not setup to test them.

File: apcd-cms/src/apps/utils/js/checkforBlankInputs.js
Comment (wesleyboar) [reply apcd-cms/src/apps/utils/js/checkforBlankInputs.js:1]: I got it working:
- #202

**Review by edmondsgarrett — APPROVED** [review]
LGTM!

**Review by wesleyboar — APPROVED** [review]

**Pull request merged**